messaging/pubnub.go

```diff
@@ -401,20 +393,6 @@ func NewPubnub(publishKey string, subscribeKey string, secretKey string, cipherK
 	return newPubnub
 }
 
-var once sync.Once
-
-// initLogging initaites the log file if loggingEnabled is true
-func initLogging() {
-	onceBody := func() {
-		infoLogger = log.New(logfileWriter, "", log.Ldate|log.Ltime|log.Lshortfile)
-		infoLogger.Printf("****************************************")
-	}
-	if (loggingEnabled) && (logfileWriter != nil) {
-		once.Do(onceBody)
-	} else {
-		infoLogger = log.New(ioutil.Discard, "", log.Ldate|log.Ltime|log.Lshortfile)
```

Comment:
This is where a race condition could occur under when the global `loggingEnabled` var is false
1. Existing instance uses it's logger (which points to the global `infoLogger`)
1.  New instance is created, reassigning the pointer to a new Logger. 

Pointer assignment in Go is not atomic, so this was a problem.